pkg/kubectl/cmd/explain.go

```diff
@@ -62,13 +62,15 @@ func NewExplainOptions(streams genericclioptions.IOStreams) *ExplainOptions {
 // NewCmdExplain returns a cobra command for swagger docs
 func NewCmdExplain(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
 	o := NewExplainOptions(streams)
+	validArgs := cmdutil.ValidArgList(f)
```

Comment:
I'm not sure this method of hardcoding resource names into commands is something we should be extending

Reply:
I agree, hardcoding that list isn't really correct. Using the `api-resources` command to see what resources the server you are talking to has is more correct (but doesn't work well for static help and autocomplete)